### PR TITLE
Lookback window: explainer & validator

### PR DESCRIPTION
**Pull request by agarant**
Update the explainer and header validator according to spec changes introduced in #914.

### CODE REVIEW

File: EVENT.md [hunk 1]
```diff
@@ -569,6 +572,18 @@ will be created.
 If the filters match for multiple event triggers, the first matching event
 trigger is used.
 
+#### Lookback window
+
+Lookback window is supported with an optional reserved keyword
+`_lookback_window` which can be added to trigger's filters.
+
+Unlike other filter keys whose values must be a list of strings. The
```
Comment (apasel422) [inline EVENT.md:580]:
```suggestion
Unlike other filter keys whose values must be a list of strings, the
```

File: EVENT.md [hunk 2]
```diff
@@ -569,6 +572,18 @@ will be created.
...
+`_lookback_window` key value must be a positive integer which represents a
```
Comment (apasel422) [inline EVENT.md:581]:
```suggestion
`_lookback_window` value must be a positive integer that represents a positive
```

File: EVENT.md [hunk 3]
```diff
@@ -569,6 +572,18 @@ will be created.
...
+duration in seconds.
+
+When available on a filter, the duration since the source was registered must be
+smaller or equal to the parsed lookback window duration for the filter to match.
```
Comment (apasel422) [inline EVENT.md:585]:
```suggestion
less than or equal to the parsed lookback window duration for the filter to match.
```

File: header-validator/src/validate-json.ts [hunk 4]
```diff
@@ -296,12 +296,21 @@ const filterData = () =>
       state.error('is prohibited because it is implicitly set')
       return
     }
+    if (filter === '_lookback_window') {
+      state.error('is prohibited because it would be ignored')
```
Comment (apasel422) [inline header-validator/src/validate-json.ts:300]: Is there a reason to use a different message for this compared to `source_type`?

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: header-validator/src/validate-json.ts (hunk 4 above)
Comment (agarant) [reply header-validator/src/validate-json.ts:300]: The idea was that we are not really comparing to a generated value, but we could probably consider it to be implicitly set. I updated.

**Review by apasel422 — APPROVED** [review]

**Review by akashnadan — COMMENTED** [review]
lgtm

**Pull request merged**